<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SocketCanInterfacePage</class>
 <widget class="QWidget" name="SocketCanInterfacePage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>670</width>
    <height>496</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QCheckBox" name="cbSocketCanConfigured">
   <property name="geometry">
    <rect>
     <x>1</x>
     <y>51</y>
     <width>411</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>let cangaroo configure this interface</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Driver:</string>
   </property>
  </widget>
  <widget class="QWidget" name="wSocketCanOptions" native="true">
   <property name="geometry">
    <rect>
     <x>91</x>
     <y>81</y>
     <width>141</width>
     <height>90</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>9</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QCheckBox" name="cbSocketCanListenOnly">
      <property name="text">
       <string>Listen only</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="cbSocketCanTripleSampling">
      <property name="text">
       <string>Triple Sampling</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="cbSocketCanCanFD">
      <property name="text">
       <string>enable CanFD</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="wSocketCanTiming" native="true">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>91</x>
     <y>181</y>
     <width>516</width>
     <height>291</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_8">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="wSocketCanTimingMode" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <property name="topMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QRadioButton" name="rbSocketCanAutomaticTiming">
         <property name="text">
          <string>automatic</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rbSocketCanManualTiming">
         <property name="text">
          <string>manual</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>310</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QStackedWidget" name="stackedSocketCanTiming">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="pageSocketCanAutoTiming">
       <widget class="QWidget" name="wSocketCanAutoTimingCanFd" native="true">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>90</y>
          <width>361</width>
          <height>80</height>
         </rect>
        </property>
        <widget class="QComboBox" name="cbSocketCanFdBitrate">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>0</y>
           <width>171</width>
           <height>23</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_23">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>5</y>
           <width>111</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>FD Bitrate:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_22">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>40</y>
           <width>121</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>FD Sample Point:</string>
         </property>
        </widget>
        <widget class="QSlider" name="slSocketCanFdSamplePoint">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>40</y>
           <width>170</width>
           <height>18</height>
          </rect>
         </property>
         <property name="maximum">
          <number>999</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="laSocketCanFdSamplePoint">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>50</y>
           <width>170</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>0.0%</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="widget_11" native="true">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>0</y>
          <width>361</width>
          <height>80</height>
         </rect>
        </property>
        <widget class="QLabel" name="label_20">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>5</y>
           <width>59</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Bitrate:</string>
         </property>
        </widget>
        <widget class="QComboBox" name="cbSocketCanBitrate">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>0</y>
           <width>170</width>
           <height>23</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_21">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>40</y>
           <width>91</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Sample Point:</string>
         </property>
        </widget>
        <widget class="QLabel" name="laSocketCanSamplePoint">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>50</y>
           <width>170</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>0.0%</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QSlider" name="slSocketCanSamplePoint">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>40</y>
           <width>170</width>
           <height>18</height>
          </rect>
         </property>
         <property name="maximum">
          <number>999</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::NoTicks</enum>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QWidget" name="pageSocketCanManualTiming">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>30</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QWidget" name="widget_6" native="true">
          <layout class="QFormLayout" name="formLayout_2">
           <property name="horizontalSpacing">
            <number>16</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QLabel" name="laSocketCanTQ">
             <property name="text">
              <string>TQ:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="spSocketCanTQ">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string> ns</string>
             </property>
             <property name="maximum">
              <number>999999</number>
             </property>
             <property name="singleStep">
              <number>5</number>
             </property>
             <property name="value">
              <number>125</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="laSocketCanPropSeg">
             <property name="text">
              <string>Prop-Seg:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="laSocketCanPhaseSeg1">
             <property name="text">
              <string>Phase Seg 1:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="laSocketCanPhaseSeg2">
             <property name="text">
              <string>Phase Seg 2:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QCheckBox" name="cbSocketCanSJW">
             <property name="text">
              <string>SJW:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="spSocketCanPropSeg">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string/>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>6</number>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="spSocketCanPhaseSeg1">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string/>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>7</number>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="spSocketCanPhaseSeg2">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string/>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>2</number>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QSpinBox" name="spSocketCanSJW">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string/>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>1</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="wSocketCanFdManualTiming" native="true">
          <layout class="QFormLayout" name="formLayout_3">
           <property name="horizontalSpacing">
            <number>16</number>
           </property>
           <property name="leftMargin">
            <number>10</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QLabel" name="laSocketCanTQ_2">
             <property name="text">
              <string>FD TQ:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="spSocketCanFdTQ">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string> ns</string>
             </property>
             <property name="maximum">
              <number>999999</number>
             </property>
             <property name="singleStep">
              <number>5</number>
             </property>
             <property name="value">
              <number>125</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="laSocketCanPropSeg_2">
             <property name="text">
              <string>FD Prop-Seg:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="laSocketCanPhaseSeg1_2">
             <property name="text">
              <string>FD Phase Seg 1:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="laSocketCanPhaseSeg2_2">
             <property name="text">
              <string>FD Phase Seg 2:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QCheckBox" name="cbSocketCanFdSJW">
             <property name="text">
              <string>FD SJW:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="spSocketCanFdPropSeg">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string/>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>6</number>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="spSocketCanFdPhaseSeg1">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string/>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>7</number>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="spSocketCanFdPhaseSeg2">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string/>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>2</number>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QSpinBox" name="spSocketCanFdSJW">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="suffix">
              <string/>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>8</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>1</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>48</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget_9" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>80</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QCheckBox" name="cbSocketCanRestart">
         <property name="text">
          <string>Automatic restart after bus off condition:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_10" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <property name="leftMargin">
           <number>20</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QSpinBox" name="spSocketCanRestartTime">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="suffix">
             <string> ms</string>
            </property>
            <property name="maximum">
             <number>999999</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>352</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>181</y>
     <width>45</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Timing:</string>
   </property>
  </widget>
  <widget class="QLabel" name="laInterfaceName">
   <property name="geometry">
    <rect>
     <x>64</x>
     <y>21</y>
     <width>231</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>TextLabel</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>21</y>
     <width>58</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Interface:</string>
   </property>
  </widget>
  <widget class="QLabel" name="laInterfaceDriver">
   <property name="geometry">
    <rect>
     <x>64</x>
     <y>0</y>
     <width>231</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>TextLabel</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_17">
   <property name="geometry">
    <rect>
     <x>31</x>
     <y>91</y>
     <width>59</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>Options:</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
